core/api/serializers.py

```
@@ -2,11 +1,208 @@
+                    origin = Contribution_origines.objects.create(**origin_data)
+                    contribution.origine.add(origin)
+
+        if notes_data:
+            for note_data in notes_data:
+                if note_data:  # Check if note data is not None
+                    note = Contribution_notes.objects.create(**note_data)
+                    contribution.notes.add(note)
+
+        return contribution
+
+    def update(self, instance, validated_data):
```

Ruff (0.8.2)

`update` is too complex (14 > 10)

(C901)